scripts/core/editor3/components/MultipleHighlights.jsx

```diff
@@ -114,105 +91,146 @@ export class MultipleHighlights extends React.Component {
             return false;
         }
 
-        if (this.props.editorState.getSelection().isCollapsed()) {
+        if (newEditorState.getSelection().isCollapsed()) {
             return false;
         }
 
         // selection is expanded to include edges
         // so you can't add a highlight right next to another
         const selection = expandDraftSelection(
-            this.props.editorState.getSelection(),
-            this.props.editorState,
+            newEditorState.getSelection(),
+            newEditorState,
             1,
             1,
             true
         );
 
-        return getDraftCharacterListForSelection(this.props.editorState, selection)
+        return getDraftCharacterListForSelection(newEditorState, selection)
             .some(characterHasAHighlightOfTheSameType.bind(this)) === false;
     }
 
-    exportState() {
-        let state = {...this.state};
+    saveHighlights({newEditorState: editorState, newHighlightsState}, applyChanges) {
+        let newEditorState = this.props.setHighlightsState(editorState, newHighlightsState);
 
-        // in order to support changing highlight styles, styleMap is not exported
-        // instead it's constructed from `highlightsData` and `availableHighlights` on `getInitialState`
-        delete state.highlightsStyleMap;
-        return state;
+        if (applyChanges) {
+            this.props.onHighlightChange(newEditorState, newHighlightsState);
+        }
+
+        return newEditorState;
     }
 
-    addHighlight(highlightType, highlightData) {
-        // NICE-TO-HAVE: remove associated data when highlights are removed by deleting characters in the editor
+    initHighlights(editorState) {
```

Comment:
`Init` is usually used as a constructor - called a single time for the lifecycle of the object. This is not how it's used here, so a better name might help IMO. How about `getHighlights`?